sanktiohaut nappaavat urakkaan liittyvät sanktiot urakan päättymisen …

jlansineva opened the pull request:
…jälkeenkin

jlansineva committed:
sanktiohaut nappaavat urakkaan liittyvät sanktiot urakan päättymisen jälkeenkin

jlansineva committed:
Merge branch 'develop' into VHAR-5849

jlansineva committed:
WIP

jlansineva committed:
castit sql:n mukaan

jlansineva committed:
Merge branch 'develop' into VHAR-5849

jlansineva committed:
testidataan sanktio urakan päättymisen jälkeen

jlansineva committed:
testit tsekkaamaan urakanjälkeisen sanktion haku mukaan urakan viimeiseen kuukauteen

jlansineva committed:
require mukaan, toimi jos ajetaan kaikki testit, mutta lein test *yksittainen-ns* ei toiminut. nyt toimii

jlansineva committed:
Merge branch 'VHAR-5849' of github.com:finnishtransportagency/harja into VHAR-5849

jlansineva committed:
Merge branch 'develop' into VHAR-5849

jlansineva committed:
tilannekuvatest - laatupoikkeamien määrä bump

jlansineva committed:
housekeeping -> lisätty require pprintille

jlansineva committed:
laadunseurannan testin päivitys - kuulumaton sanktio id bump

rckopteri commented on src/clj/harja/palvelin/raportointi/raportit/sanktiot.sql:
```diff
@@ -37,7 +37,14 @@ WHERE ((:urakka::INTEGER IS NULL AND u.urakkanro IS NOT NULL) OR u.id = :urakka)
+          (CASE 
+                date_part('year', :loppu::date)::integer = date_part('year', u.loppupvm)::integer 
+                AND date_part('month', :loppu::date)::integer = date_part('month', u.loppupvm)::integer
+           WHEN TRUE THEN s.perintapvm > u.loppupvm 
```
Mulle ei ole tuttu tällainen rakenne, että CASE:n jälkeen ei tule WHEN vaan ehto. Eli tämäkin varmaan toimii, mutta onko olemassa jokin syy, miksi CASE ei olisi tämän muotoinen:
(CASE 
WHEN
       date_part('year', :loppu::date)::integer = date_part('year', u.loppupvm)::integer 
                AND date_part('month', :loppu::date)::integer = date_part('month', u.loppupvm)::integer 
AND s.perintapvm > u.loppupvm
          THEN TRUE
ELSE FALSE
           END)

jlansineva replied:
joo, tää on mullekin uusi rakenne, mitä ajattelin kokeilla. Bongasin sen tuolta postgresqln dokkareista, jossa mainittiin, että se toimii switch casen tapaan mikä tässä vaikutti ihan näppärältä, kun kiinnostaa vain tuo yksi tapaus tuossa ja sen tulos. Ei sinällään muuta syytä sen käyttöön.

rckopteri commented on test/clj/harja/palvelin/raportointi/sanktioraportti_test.clj:
```diff
@@ -128,3 +128,88 @@
+
+(deftest raportin-suoritus-urakan-jalkeen-tulleilla-sanktioilla-toimii
+  (let [urakalla-sanktiot-tulee-mukaan-viimeisella-hoitokaudella
+        (kutsu-palvelua (:http-palvelin jarjestelma)
```
Haku näyttäisi olevan aika identtinen useamman kerran. Olisi helpompi lukea koodia, jos haku olisi funktion takana ja muuttuvat tiedot parametrisoituina.

rckopteri commented on test/clj/harja/palvelin/raportointi/sanktioraportti_test.clj:
```diff
@@ -128,3 +128,88 @@
+                        (= (first %) "Ryhmä C, sakot yht."))
+                    (= (second %) 777M))
+                 (as-> urakalla-sanktiot-tulee-mukaan-viimeisella-hoitokaudella u
+                   (nth u 2)
```
Meillä on olemassa helppereitä raporttien testaamiseksi: harja.palvelin.raportointi.testiapurit
Ehkä voisit kurkata, olisiko tuolla jotain, mitä käyttää apuna?

jlansineva committed:
testikoodin siistimistä

rckopteri commented on src/cljc/harja/tyokalut/yleiset.cljc:
```diff
@@ -52,8 +53,8 @@
    (let [f (fn [prefix object]
              (print prefix)
-             (clojure.pprint/pprint object)
+             (pp/pprint object)
```
Onkohan ClojureScriptin puolella prettyprinttiä olemassa? Jotenki fiilis, että tämä täytyis wräpätä clj koodiksi tai muuten tulee käynnistyksessä virhettä. (tämä ei toki oli sun tekemä koodi, vain siistit sitä, mutta jos sulla on energiaa, niin ei haittaisi siistiä)

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into VHAR-5849

jarnovayrynen committed:
Palauta sanktiopalvelusta Java dateja eikä SQL-timestampejä

Hae sanktiot SQL:stä avaimien avulla eikä positionaalisten parametrien.

jarnovayrynen committed:
Päättymisen jälkeiset sanktiot viimeiselle hoitokaudelle myös näkymässä

Hae urakan päättymisen jälkeiset sanktiot viimeiselle hoitokaudelle myös Laadunseuranan Sanktiot-näkymässä jotta toteutus yhdenmukainen raportin kanssa.

jarnovayrynen committed:
Luo testi joka testaa päättymisen jälkeisen sanktion haun

jarnovayrynen committed:
Päivitä tilannekuvan lp-count koska uusi rivi testidataan
